node/full_node_test.go

```diff
@@ -303,6 +363,46 @@ func createAggregatorWithPersistence(ctx context.Context, dbPath string, dalc *d
 	return fullNode, app
 }
 
+func createAggregatorWithApp(ctx context.Context, app abci.Application, voteExtensionEnableHeight int64, t *testing.T) (Node, crypto.PrivKey) {
```

Comment:
may be you can use `createNodes` from the `full_node_integration_test.go` with additional parameter?

Reply:
Answering also to @MSevey:
Yes, I was thinking about refactoring this, but probably we should have `Options` pattern to be able to use enforce specific behaviors without having too many arguments. There's also difference in returned params.

Or maybe we could/should introduce some kind of builder that can be configured and also return various bits (like p2p key, signing key, genesis, application, etc.

Reply:
yea agreed. let's just create a follow up issue as it will create too much blot to this PR. 
But definitely time to refactor some of this test code.

Reply:
https://github.com/rollkit/rollkit/issues/1648